int32 id
string data
bool last
